### ISSUE TITLE
ci: lint scripts with shellcheck

### ISSUE THREAD

**Issue by katexochen** [confidential-containers/cloud-api-adaptor#938]
Issues like #936 can be prevented by linting shell scripts with [shellcheck](https://github.com/koalaman/shellcheck) on PRs.

If people agree, I'll add a workflow to the CI for linting scripts.

**Comment by bpradipt** [comment]
@katexochen it will be a good addition. 

@stevenhorsman @wainersm @surajssd  and others hope you are fine with it as well.

**Comment by surajssd** [comment]
Of course it will be a good addition. Thanks @katexochen

**Comment by jtumber-ibm** [comment]
As long as it doesn't initially block PRs and we work on fixing them as separate issues I think this would be helpful as we probably have quite a big backlog of issues (237 from a quick count)

**Comment by stevenhorsman** [comment]
Yeah, this is a great idea, but as JT says maybe we don't make it required until we've been able to fix up the issues to avoid blocking all PRs?

**Comment by katexochen** [comment]
I'll try to fix most issues upfront.
